### PR TITLE
Dplyr 1.0

### LINKED ISSUE #1783: Warning: `arrange_()` is deprecated as of dplyr 0.7.0. Please use `arrange()` instead.
I update dplyr to 1.0.0 and the following message appears:

`Warning: `arrange_()` is deprecated as of dplyr 0.7.0. Please use `arrange()` instead. See vignette('programming') for more help`

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #1783, #1825, and #1799

### CODE REVIEW

File: R/ggplotly.R [hunk 1]
```diff
@@ -1112,7 +1112,7 @@ gg2list <- function(p, width = NULL, height = NULL,
   # translate group aesthetics to data attributes
   return_dat <- Map(function(x, y) {
     if (is.null(y[["group"]])) return(x)
-    dplyr::group_by_(x, y[["group"]])
+    dplyr::group_by(x, !!rlang::as_quosure(y[["group"]]))
```
Comment (cpsievert) [inline R/ggplotly.R:1115]: TODO: should this be `sym()` instead of `as_quosure()`?

Comment (cpsievert) [reply R/ggplotly.R:1115]: Ah, no, make sense because `y` is formula

File: DESCRIPTION [hunk 2]
```diff
@@ -75,7 +75,7 @@ Suggests:
     forcats,
```
Comment (schloerke) [inline DESCRIPTION:75]: Need to add a version requirement for `  dplyr (>= 1.0.0)`

Comment (cpsievert) [reply DESCRIPTION:75]: 1.0.0 isn't actually required

Comment (schloerke) [reply DESCRIPTION:75]: The comment is for `dplyr`. I can't comment up there

Comment (cpsievert) [reply DESCRIPTION:75]: I know, I don't want to require dplyr 1.0, it should work with older versions too

File: R/onLoad.R [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  # Always register hook in case pkg is loaded at some
+  # point the future (or, potentially, but less commonly,
+  # unloaded & reloaded)
+  setHook(
+    packageEvent(pkg, "onLoad"),
+    function(...) {
+      registerS3method(generic, class, fun, envir = asNamespace(pkg))
+    }
+  )
+}
```
Comment (schloerke) [inline R/onLoad.R:41]: Consider copying from `vctrs`: https://github.com/r-lib/vctrs/blob/df8a659135149fb7c97475c80b4f0e3703a5d784/R/register-s3.R#L49-L105

Docs: https://vctrs.r-lib.org/reference/s3_register.html

Via: https://github.com/r-lib/roxygen2/issues/623

File: R/plotly_data.R [hunk 4]
```diff
@@ -104,116 +104,161 @@ highlight_key <- function(...) {
...
 #' @rdname plotly_data
-#' @export
 groups.plotly <- function(x) {
-  dplyr::groups(plotly_data(x))
+  groups(plotly_data(x))
 }
 
 #' @rdname plotly_data
-#' @export
 ungroup.plotly <- function(x, ...) {
```
Comment (schloerke) [inline R/plotly_data.R:117]: This is now unexported?  I'm fine with it as long as it still works

File: R/utils.R [hunk 5]
```diff
@@ -139,7 +139,7 @@ crosstalk_key <- function() ".crossTalkKey"
 # arrange data if the vars exist, don't throw error if they don't
 arrange_safe <- function(data, vars) {
   vars <- vars[vars %in% names(data)]
-  if (length(vars)) dplyr::arrange_(data, .dots = vars) else data
+  if (length(vars)) dplyr::arrange(data, !!!rlang::syms(vars)) else data
```
Comment (schloerke) [inline R/utils.R:142]: Does `rlang` automatically import `!!!`? Does it need to be imported/accessed manually?

File: R/plotly_data.R (hunk 4 above)
Comment (cpsievert) [reply R/plotly_data.R:129]: Yea, the method is now registered in `.onLoad()` instead

File: R/utils.R (hunk 5 above)
Comment (cpsievert) [reply R/utils.R:142]: I did end up importing, but I don't think it's necessary:

```r
> rlang::`!!`
function (x) 
{
    abort("`!!` can only be used within a quasiquoted argument")
}
```

**Pull request merged**